File: CS/OfficeUniversalEx.cs
using DevExpress.Skins;
using DevExpress.Utils.Drawing;
using DevExpress.XtraBars.Painters;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.Drawing;
using DevExpress.XtraBars.Ribbon.ViewInfo;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace OfficeUniversalExample {
    public class RibbonControlEx : RibbonControl {
        protected override DevExpress.XtraBars.Ribbon.ViewInfo.RibbonViewInfo CreateViewInfo() {
            return new RibbonViewInfoEx(this);
        }
    }

    public class RibbonViewInfoEx : RibbonViewInfo {
        public RibbonViewInfoEx(RibbonControl ribbon) : base(ribbon) { }
        protected override RibbonPageHeaderViewInfo CreateHeaderInfo() {
            return new RibbonPageHeaderViewInfoEx(this);
        }
        protected override RibbonCaptionViewInfo CreateCaptionInfo() {
            return new RibbonCaptionViewInfoEx(this);
        }
        protected override int UniversalOfficeApplicationButtonLeftIndent {
            get { return base.UniversalOfficeApplicationButtonLeftIndent + Caption.ContentBounds.X; }
        }
        protected override int CalcMinHeight() {
            return base.CalcMinHeight() - Caption.Bounds.Height + TopIndent + TopIndent;
        }
        public static int TopIndent { get { return 5; } }
    }

    public class RibbonCaptionViewInfoEx : RibbonCaptionViewInfo {
        public RibbonCaptionViewInfoEx(RibbonViewInfo info) : base(info) { }
        protected override ObjectPainter CreateCaptionPainter() {
            return new RibbonCaptionPainterEx();
        }
    }

    public class RibbonPageHeaderViewInfoEx : RibbonPageHeaderViewInfo {
        public RibbonPageHeaderViewInfoEx(RibbonViewInfo info) : base(info) { }
        protected override Rectangle CalcBounds() {
            Rectangle r = base.CalcBounds();
            Rectangle res = new Rectangle(r.X, RibbonViewInfoEx.TopIndent, ViewInfo.Caption.ContentBounds.Width, r.Height + RibbonViewInfoEx.TopIndent);
            if(Ribbon.FindForm() == null || Ribbon.FindForm().WindowState != FormWindowState.Maximized) return res;
            res.Y += RibbonViewInfoEx.TopIndent;
            return res;
        }
        protected override Rectangle CalcAvailableHeaderRect() {
            Rectangle rect = base.CalcAvailableHeaderRect();
            rect.Width -= PageHeaderItemsBounds.Width;
            return rect;
        }
    }

    public class RibbonCaptionPainterEx : RibbonCaptionPainter {
        public override void DrawCaption(ObjectInfoArgs e) { }
        protected override void DrawPageHeaderBackground(ObjectInfoArgs e, bool upperPart) { }
    }
}

File: VB/OfficeUniversalEx.vb
Imports DevExpress.Skins
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraBars.Painters
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.Drawing
Imports DevExpress.XtraBars.Ribbon.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace OfficeUniversalExample
    Public Class RibbonControlEx
        Inherits RibbonControl

        Protected Overrides Function CreateViewInfo() As DevExpress.XtraBars.Ribbon.ViewInfo.RibbonViewInfo
            Return New RibbonViewInfoEx(Me)
        End Function
    End Class

    Public Class RibbonViewInfoEx
        Inherits RibbonViewInfo

        Public Sub New(ByVal ribbon As RibbonControl)
            MyBase.New(ribbon)
        End Sub
        Protected Overrides Function CreateHeaderInfo() As RibbonPageHeaderViewInfo
            Return New RibbonPageHeaderViewInfoEx(Me)
        End Function
        Protected Overrides Function CreateCaptionInfo() As RibbonCaptionViewInfo
            Return New RibbonCaptionViewInfoEx(Me)
        End Function
        Protected Overrides ReadOnly Property UniversalOfficeApplicationButtonLeftIndent() As Integer
            Get
                Return MyBase.UniversalOfficeApplicationButtonLeftIndent + Caption.ContentBounds.X
            End Get
        End Property
        Protected Overrides Function CalcMinHeight() As Integer
            Return MyBase.CalcMinHeight() - Caption.Bounds.Height + TopIndent + TopIndent
        End Function
        Public Shared ReadOnly Property TopIndent() As Integer
            Get
                Return 5
            End Get
        End Property
    End Class

    Public Class RibbonCaptionViewInfoEx
        Inherits RibbonCaptionViewInfo

        Public Sub New(ByVal info As RibbonViewInfo)
            MyBase.New(info)
        End Sub
        Protected Overrides Function CreateCaptionPainter() As ObjectPainter
            Return New RibbonCaptionPainterEx()
        End Function
    End Class

    Public Class RibbonPageHeaderViewInfoEx
        Inherits RibbonPageHeaderViewInfo

        Public Sub New(ByVal info As RibbonViewInfo)
            MyBase.New(info)
        End Sub
        Protected Overrides Function CalcBounds() As Rectangle
            Dim r As Rectangle = MyBase.CalcBounds()
            Dim res As New Rectangle(r.X, RibbonViewInfoEx.TopIndent, ViewInfo.Caption.ContentBounds.Width, r.Height + RibbonViewInfoEx.TopIndent)
            If Ribbon.FindForm() Is Nothing OrElse Ribbon.FindForm().WindowState <> FormWindowState.Maximized Then
                Return res
            End If
            res.Y += RibbonViewInfoEx.TopIndent
            Return res
        End Function
        Protected Overrides Function CalcAvailableHeaderRect() As Rectangle
            Dim rect As Rectangle = MyBase.CalcAvailableHeaderRect()
            rect.Width -= PageHeaderItemsBounds.Width
            Return rect
        End Function
    End Class

    Public Class RibbonCaptionPainterEx
        Inherits RibbonCaptionPainter

        Public Overrides Sub DrawCaption(ByVal e As ObjectInfoArgs)
        End Sub
        Protected Overrides Sub DrawPageHeaderBackground(ByVal e As ObjectInfoArgs, ByVal upperPart As Boolean)
        End Sub
    End Class
End Namespace
